### DISCUSSION TITLE
DateTimeStump, C array

### DISCUSSION THREAD

**Discussion by FalloutST** [fnc12/sqlite_orm#757] [Q&A] [answered]
Hi there,
I think to add char[size] to access C structures,
How to add datetimestump?...

**Comment by fnc12** [comment]
@FalloutST how would you implement it in raw sqlite?

**Reply by FalloutST** [reply]
char[size]  as text
datetime as ulong with convertion to  std::crono

**Reply by fnc12** [reply]
looks like you want to store date as dynamic C char array. It is not supported in `sqlite_orm` but you can add this support by creating `statement_binder` and `row_extractor` specialization for your type just like all other specializations are created. My personal advice - try avoiding using C type. Use classes instead in C++

**Reply by FalloutST** [reply]
I have already tried.
Unfortunately, not achieved success, yet.
At that moment, it bypassed this using different structures, to access database and storage Shared Memory.
Thanks for the direction, I will try to do it later.
It also lacks the ability to use STD :: Array, or another type of data stored not in heap, to store:
struct Foo {
char name [30];
...
...
}

**Reply by FalloutST** [reply]
I store date as ulong msecsSE, and access it as std::crono

**Reply by fnc12** [reply, marked as answer]
you may add bindings for `std::array`. There are no bindings for `std::array` out of box cause nobody knows how everybody wants to encode some array data into column: json, xml or something custom. When you say 'I miss %some_field_type% support' you need to understand how would you store it using raw sqlite first: how would you insert it (as string, as blob, if string then what format) and how would you select it. I don't know how you want to do it. Moreover somebody else will want to using `std::array` in a different way
